python/dataproc_templates/hive/util/hive_ddl_extractor.py

```diff
@@ -0,0 +1,170 @@
+                ddl_table.registerTempTable("metadata")
+                show_extended.registerTempTable("show_extended")
+                info=spark.sql("select information from show_extended")
+                partition_check=show_create.filter(show_create["createtab_stmt"].contains("PARTITIONED")).head(1)
+                partition_string=""
+                if partition_check:
+                    partition=show_create.first()[0].split("PARTITIONED BY (")[1].split(")")[0]
+                    partitions=partition.split(",")
+                    for part in partitions:
+                        partition_type=spark.sql("select data_type from metadata where col_name="+"'"+part.strip()+"'").first()[0]
+                        partition_string=partition_string+" "+part+" "+partition_type+","
+                    partition_string=partition_string[:-1]
+                    partition_by="PARTITIONED BY ("+partition_string+")\n"
+                    first_part=show_create.first()[0].split(partitions[0])[0][:-5].split(t.name+"`")[1:]
+                    first_part="".join(first_part)+")\n"
+                else:
+                    partition_by=")\n"
+                    first_part=show_create.first()[0].split(t.name)[1:]
+                    first_part=show_create.first()[0].split(')')[0].split(t.name+"`")[1:]
+                    first_part="".join(first_part)+")\n"
+                db_name=spark.sql("select data_type from metadata where col_name='Database'").first()[0]
+                table_name=spark.sql("select data_type from metadata where col_name='Table'").first()[0]
+                hdfs_file_path=info.first()[0].split("Location: ")[1].split("\n")[0]
+                format=show_create.first()[0].split("USING ")[1].split("\n")[0]
+                
+                #partition and format
+                initial="CREATE TABLE IF NOT EXISTS "+t.name
+                storage_format=format
+                location_path=gcs_staging_path+hdfs_file_path.split('hdfs://')[1]
+                metadata.append([db_name,table_name,partition_string,storage_format,hdfs_file_path,location_path])
+                ddl = initial+first_part+partition_by+";\n"
+                ddls = ddls + ddl
+            rdd=spark.sparkContext.parallelize(metadata)
+            metadata_df=rdd.toDF(columns)
+            # Write metadata to BQ
+            metadata_df.write \
+                .format(constants.FORMAT_BIGQUERY) \
+                .option(constants.TABLE, bigquery_dataset+'.'+bigquery_table) \
+                .option(constants.TEMP_GCS_BUCKET, gcs_working_directory) \
+                .mode(constants.OUTPUT_MODE_APPEND) \
+                .save()
+        else:
+            logger.error("This database does not exist")
+            exit (1)
+
+        # Write the DDLs to GCS
+        ddls_to_rdd = spark.sparkContext.parallelize([ddls])
+        WriteToCloud(ddls_to_rdd, gcs_working_directory, gcs_target_path)
```

Comment:
These 2 lines should be inside the dbCheck if statement

Reply:
done